crates/iota-data-ingestion/src/workers/blob.rs

```diff
@@ -24,10 +30,91 @@ pub struct BlobWorker {
 impl BlobWorker {
     pub fn new(config: BlobTaskConfig) -> Self {
         Self {
-            remote_store: create_remote_store_client(config.url, config.remote_store_options, 10)
-                .expect("failed to create remote store client"),
+            remote_store: create_remote_store_client_with_ops(
+                config.url,
+                config.remote_store_options,
+                config.timeout_secs,
+                RetryConfig {
+                    max_retries: 10,
+                    retry_timeout: Duration::from_secs(config.timeout_secs + 1),
+                    ..Default::default()
+                },
+            )
+            .expect("failed to create remote store client"),
         }
     }
+
+    /// Uploads a Checkpoint blob to the Remote Store.
+    ///
+    /// If the blob size exceeds the configured `CHUNK_SIZE`,
+    /// it uploads the blob in parts using multipart upload.
+    /// Otherwise, it uploads the blob directly.
+    async fn upload_blob(&self, bytes: Vec<u8>, chk_seq_num: u64, location: Path) -> Result<()> {
+        if bytes.len() > CHUNK_SIZE {
+            return self
+                .upload_blob_multipart(bytes, chk_seq_num, location)
+                .await;
+        }
+
+        self.remote_store
+            .put(&location, Bytes::from(bytes).into())
```

Comment:
why don't we have here the same log message with checkpoint number and time elapsed?
you could even calculate the average upload speed.

Reply:
The `iota-data-ingestion-core` already provides a [similar log](https://github.com/iotaledger/iota/blob/develop/crates/iota-data-ingestion-core/src/worker_pool.rs#L84), the multipart logs are only tied to that part as I wanted to have more detailed information for those checkpoints which could have multiple GBs